src/agent/agent.js

```diff
@@ -76,30 +87,37 @@ export class Agent {
         return this.bot.chat(message);
     }
 
-    async handleMessage(source, message) {
+    async handleMessage(source, message, self_prompt=false) {
```

Comment:
Mixing code for self prompting into this loop feels messy. And I feel like non of the uses of this `self_prompt` variable are necessary either with a more generic argument or by moving the logit elsewhere.